src/NGen.cpp

```diff
@@ -541,11 +531,20 @@ int main(int argc, char* argv[]) {
     // T-ROUTE data storage
     std::unordered_map<std::string, int> nexus_indexes;
 #if NGEN_WITH_ROUTING
-    size_t nexus_collection_size = nexus_collection->get_size();
-    for (int i = 0; i < nexus_collection_size; ++i) {
-        auto feature = nexus_collection->get_feature(i);
-        std::string feature_id = feature->get_id();
-        nexus_indexes[feature_id] = i;
+    // create sorted indexes for possible MPI
+    {
+        std::vector<std::string> routing_nexus_ids;
+        for (int i = 0; i < nexus_collection->get_size(); ++i) {
```

Comment:
A note either to you or myself: check this back against `nexus_subset_ids` in the parallel case, where the set of features is partitioned among processes.